src/public-assets/js/dependent-dropdown.js

```diff
@@ -0,0 +1,19 @@
+var Dropdown = (function ($) {
```

Comment:
@kenny-wealth your approach requires the API request & HTML to be structured in a specific way if it's going to work. I think this could be more generic so it can serve more use cases. We can discuss this further if you need more clarification.